### ISSUE TITLE
Question about usage

### ISSUE THREAD

**Issue by ldelossa** [gogo/protobuf#416]
Hello,

Will using GoGo Protobuf and it's non-standard directives in the proto files make generated protos for other languages break?

**Comment by awalterschulze** [comment]
Tl;dr Gogoprotobuf is not doing anything non standard or against the protobuf spec and aims to be compliant, exactly to be interoperable with other languages.

Importing gogoproto is the only requirement.
Technically this should be supported by all languages.
I have personally had success with Java.
I have heard about success with C++.

I have heard about some problems with PHP, because its implementation only supports proto3, which does not support proto2 extensions.
Gogoproto technically only extends the descriptor which is still allowed by proto3, but I think php devs ran out of time to cover this case.
You are still able to use protoc-gen-gogofast, etc without importing gogoproto.

I hope that helps answer your question?

**Comment by ldelossa** [comment]
So my main goal is to create Proto definitions and share them between Go
GRPC server and python GRPC clients. If I generate the Proto with the gogo
protoc conpile I will need to do the same for the python code generation,
I.e I can’t use the original protoc tool with “python_out” ?

**Comment by awalterschulze** [comment]
Both languages use protoc as far as I can tell.
protoc itself, understands extensions and outputs a FileDescriptorSet which is consumed by python_out.
There is a small chance that the python code generator (python_out) does not understand extensions, but I am unaware of this short coming.
You should be able to use protoc --python_out of the python code generator has a bug.

**Comment by ldelossa** [comment]
Thanks a ton! Answers my questions

**Comment by awalterschulze** [comment]
I am going to close this issue, since your question is answered, but we can open it again if you have more questions.
